src/ContractForm.js

```diff
@@ -73,8 +72,22 @@ class ContractForm extends Component {
   }
 
   render() {
+    // If a render prop is given, have displayData rendered from that component
+    if (this.props.render) {
+      // calls render with array of input names and event handler
+      return this.props.render(
+        this.inputs.reduce((map, input) => {
+          map[input.name] = input.ref;
+          return map;
+        }, {}),
+        this,
```

Comment:
This also seems like a lot to pass down, I'd prefer a more targeted approach to passing down props.

Reply:
Here it needs to pass something down, so that the rebder function can create form inputs out of it, and the result can still be read in the parent ContractForm. That works nice with the createRef() object.

What would you suggest as alternative approach?